### ISSUE TITLE
Speech and Language Processing

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#10978]
Speech and Language Processing<br>
<br>
https://ift.tt/TPgbYr1<br>
<br>
<br>
<br>
<div><div>
<h3>
Here's our Feb 3, 2024 release! We also expect to release Chapter 12 soon in an updated release.
</h3>
<p>
Individual chapters and updated slides are below;
<a href="https://web.stanford.edu/~jurafsky/slp3/ed3bookfeb3_2024.pdf">here is a single pdf of all the chapters in the Feb 3, 2024 release!</a> <br>
</p>
<p> <b>Feel free to use the draft chapters and slides in your classes, the resulting feedback we get from you makes the book better!</b><br>
As always, typos and comments very welcome (just email <a href="mailto:slp3edbugs@gmail.com">slp3edbugs@gmail.com</a>
and let us know the date on the draft)!
(Don't bother reporting missing refs due to cross-chapter cross-reference problems in the indvidual chapter pdfs, those are fixed in the full book draft)<br>
</p>
<h5>
We've put up a <a href="https://web.stanford.edu/~jurafsky/slp3/thanks.html">list here</a> of the amazing people who have sent so many fantastic suggestions and bug-fixes for improving the book.
We are really grateful to all of you for your help, the book would not be possible without you!
</h5>
<p>
When will the whole book be finished? Don't ask. <br>
</p>
<p>
If you need last year's Jan 2023 draft chapters, they are <a href="https://web.stanford.edu/~jurafsky/slp3/old_jan23/">here</a>;
</p>
<table>
<tbody>
<tr><td> </td></tr>
<tr>
<td></td>
<td><b>Chapter</b></td>
<td><b>Slides </b></td>

</tr>
<tr>
<td></td>
<td><b>Part I: Fundamental Algorithms</b></td>
</tr>
<tr>
<td>1:</td>
<td>Introduction</td>
</tr>
<tr>
<td>2:</td>
<td>
<a href="https://web.stanford.edu/~jurafsky/slp3/2.pdf">Regular Expressions, Text Normalization, Edit Distance</a>
</td>
<td>
2: Text Processing
[<a href="https://web.stanford.edu/~jurafsky/slp3/slides/2_TextProc_2023.pptx">pptx</a>] [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/2_TextProc_2023.pdf">pdf</a>]<br>
2: Edit Distance
[<a href="https://web.stanford.edu/~jurafsky/slp3/slides/2_EditDistance_2023.pptx">pptx</a>] [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/2_EditDistance_2023.pdf">pdf</a>]<br>
</td>
</tr>
<tr>
<td>3:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/3.pdf">N-gram Language Models</a>
</td>
<td>
3: [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/3_LM_2024.pptx">pptx</a>] [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/3_LM_2024.pdf">pdf</a>]<br>
</td>
</tr>
<tr>
<td>4:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/4.pdf"> Naive Bayes, Text Classification, and Sentiment </a>
</td>
<td>
4: [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/4_NB_2024.pptx">pptx</a>] [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/4_NB_2024.pdf">pdf</a>]<br>
</td>
</tr>
<tr>
<td>5:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/5.pdf"> Logistic Regression</a>
</td>
<td>
5: [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/5_LR_Apr_7_2021.pptx">pptx</a>] [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/5_LR_Apr_7_2021.pdf">pdf</a>]<br>
</td>
</tr>
<tr>
<td>6:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/6.pdf"> Vector Semantics and Embeddings</a>
</td>
<td>
6: [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/vectorsemantics2024.pptx">pptx</a>] [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/vectorsemantics2024.pdf">pdf</a>]
</td>
</tr>
<tr>
<td>7:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/7.pdf">Neural Networks and Neural Language Models </a>
</td>
<td>
7: [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/7_NN_Apr_28_2021.pptx">pptx</a>] [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/7_NN_Apr_28_2021.pdf">pdf</a>]
</td>
</tr>
<tr>
<td>8:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/8.pdf">Sequence Labeling for Parts of Speech and Named Entities</a>
</td>
<td>
8: (Intro only) [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/8_POSNER_intro_May_6_2021.pptx">pptx</a>] [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/8_POSNER_intro_May_6_2021.pdf">pdf</a>]
</td>
</tr>
<tr>
<td>9:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/9.pdf">RNNs and LSTMs</a> </td>
</tr>
<tr>
<td>10:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/10.pdf">Transformers and Large Language Models</a> </td>
</tr>
<tr>
<td>11:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/11.pdf">Fine-tuning and Masked Language Models</a> </td>
</tr>
<tr>
<td>12:</td>
<td> Prompting, In-Context Learning, and Instruct Tuning </td>
</tr>
<tr><td> </td></tr>
<tr>
<td></td>
<td><b>Part II: NLP Applications</b></td>
</tr>
<tr>
<td>13:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/13.pdf">Machine Translation</a> </td>
</tr>
<tr>
<td>14:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/14.pdf">Question Answering and Information Retrieval</a> </td>
</tr>
<tr>
<td>15:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/15.pdf">Chatbots and Dialogue Systems</a> </td>
<td> 15 [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/24_Dialogue_May_6_2021.pptx">pptx</a>] [<a href="https://web.stanford.edu/~jurafsky/slp3/slides/24_Dialogue_May_6_2021.pdf">pdf</a>]
</td>
</tr>
<tr>
<td>16:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/16.pdf">Automatic Speech Recognition and Text-to-Speech</a>
</td>
</tr>
<tr><td> </td></tr>
<tr>
<td></td>
<td><b>Part III: Annotating Linguistic Structure</b></td>
</tr>
<tr>
<td>17:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/17.pdf">Context-Free Grammars and Constituency Parsing</a>
</td>
</tr>
<tr>
<td>18:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/18.pdf">Dependency Parsing </a>
</td>
</tr>
<tr>
<td>19:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/19.pdf">Information Extraction: Relations, Events, and Time</a>
</td>
</tr>
<tr>
<td>20:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/20.pdf">Semantic Role Labeling and Argument Structure</a>
</td>
</tr>
<tr>
<td>21:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/21.pdf">Lexicons for Sentiment, Affect, and Connotation </a>
</td>
</tr>
<tr>
<td>22:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/22.pdf">Coreference Resolution</a>
</td>
</tr>
<tr>
<td>23:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/23.pdf">Discourse Coherence</a>
</td>
</tr>
<tr><td> </td></tr>
<tr>
<td></td>
<td><b>Appendix Chapters (will be just on the web)</b></td>
</tr>
<tr>
<td>A:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/A.pdf">Hidden Markov Models</a> </td>
</tr>
<tr>
<td>B:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/B.pdf">Spelling Correction and the Noisy Channel</a> </td>
</tr>
<tr>
<td>C:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/C.pdf">Statistical Constituency Parsing</a> </td>
</tr>
<tr>
<td>D:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/D.pdf">Context-Free Grammars</a> </td>
</tr>
<tr>
<td>E:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/E.pdf">Combinatory Categorial Grammar</a> </td>
</tr>
<tr>
<td>F:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/F.pdf">Logical Representations of Sentence Meaning</a> </td>
</tr>
<tr>
<td>G:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/G.pdf">Word Senses and WordNet</a> </td>
</tr>
<tr>
<td>H:</td>
<td> <a href="https://web.stanford.edu/~jurafsky/slp3/H.pdf">Phonetics</a> </td>
<td>
</td>
</tr>
</tbody>
</table>
</div></div><br>
<br>
<br>
<br>
<br>
<br>
via web.stanford.edu https://ift.tt/fir7LJM<br>
<br>
April 2, 2024 at 06:53PM
